trunk/SUAVE/Methods/Weights/Buildups/Common/prop.py

```diff
@@ -0,0 +1,270 @@
+    skinLength = np.sum(np.sqrt(np.sum(np.diff(box,axis=0)**2,axis=1)))
+    maxThickness = (np.amax(box[:,1])-np.amin(box[:,1]))/2
+    rootBendingMoment = SF*maxThrust/nBlades*0.75*rProp
+    m = (unidirectional_carbon_fiber().density*dx*rootBendingMoment/
+        (2*unidirectional_carbon_fiber().ultimate_shear_strength*maxThickness))+ \
+        skinLength*bidirectional_carbon_fiber().minimum_gage_thickness*dx*bidirectional_carbon_fiber().density
+    m = m*np.ones(N)
+    error = 1               # Initialize Error
+    tolerance = 1e-8        # Mass Tolerance
+    massOld = np.sum(m)
+
+#-------------------------------------------------------------------------------
+# General Structural Properties
+#-------------------------------------------------------------------------------
+
+    seg = []                        # LIST of Structural Segments
+
+    # Torsion
+
+    enclosedArea = 0.5*np.abs(np.dot(box[:,0],np.roll(box[:,1],1))-
+        np.dot(box[:,1],np.roll(box[:,0],1)))   # Shoelace Formula
+
+    # Flap Properties
+
+    box = coord                     # Box Initially Matches Airfoil
+    box = box[box[:,0]<=fwdWeb[1]]   # Trim Coordinates Aft of Aft Web
+    box = box[box[:,0]>=fwdWeb[0]]   # Trim Coordinates Fwd of Fwd Web
+    seg.append(box[box[:,1]>np.mean(box[:,1])]*chord)   # Upper Fwd Segment
+    seg.append(box[box[:,1]<np.mean(box[:,1])]*chord)   # Lower Fwd Segment
+
+    # Flap & Drag Inertia
+
+    capInertia = 0
+    capLength = 0
+
+    for i in range(0,2):
+        l = np.sqrt(np.sum(np.diff(seg[i],axis=0)**2,axis=1))   # Segment Lengths
+        c = (seg[i][1::]+seg[i][0::-1])/2                       # Segment Centroids
+
+        capInertia += np.abs(np.sum(l*c[:,1] **2))
+        capLength  += np.sum(l)
+
+    # Shear Properties
+
+    box = coord
+    box = box[box[:,0]<=fwdWeb[1]]
+    z = box[box[:,0]==fwdWeb[0],1]*chord
+    shearHeight = np.abs(z[0] - z[1])
+
+    # Core Properties
+
+    box = coord
+    box = box[box[:,0]>=fwdWeb[0]]
+    box = box*chord
+    coreArea = 0.5*np.abs(np.dot(box[:,0],np.roll(box[:,1],1))-
+        np.dot(box[:,1],np.roll(box[:,0],1)))   # Shoelace Formula
+
+    # Shear/Moment Calculations
+
+    Vz = np.append(np.cumsum(( F[0:-1]*np.diff(x))[::-1])[::-1],0)  # Bending Moment
+    Mx = np.append(np.cumsum((Vz[0:-1]*np.diff(x))[::-1])[::-1],0)  # Torsion Moment
+    My = np.append(np.cumsum(( Q[0:-1]*np.diff(x))[::-1])[::-1],0)  # Drag Moment
+
+#-------------------------------------------------------------------------------
+# Mass Calculation
+#-------------------------------------------------------------------------------
+
+    while error > tolerance:
+        CF = (SF*omega**2*
+            np.append(np.cumsum(( m[0:-1]*np.diff(x)*x[0:-1])[::-1])[::-1],0))  # Centripetal Force
+
+        # Calculate Skin Weight Based on Torsion
+
+        tTorsion = My/(2*bidirectional_carbon_fiber().ultimate_shear_strength*enclosedArea)                          # Torsion Skin Thickness
```

Comment:
unpack these

Reply:
Changed to unpack of material properties.